org/polypheny/prism/graph_frame.proto

```diff
@@ -14,10 +15,20 @@ option csharp_namespace = "Polypheny.Prism";
 package org.polypheny.prism;
 
 /*
-The GraphFrame message is a placeholder intended to represent graph frames as part of the labeled property graph model.
-As of now, the graph model is not fully implemented, making this message void of fields.
-It acts as a provision for future developments where fields related to the graph model will be added.
-Currently, this message does not contain any fields. It is reserved for future use when the graph model is more completely realized.
-*/
+ * The GraphFrame represents the non-relational result of a query in the labeled property graph model.
+ * If the query returns a relational result, such as when aggregations are used or specific properties are returned,
+ * a relational frame is used. The GraphFrame is used for results where a set of nodes, edges, or paths is returned.
+ * This selection is exclusive, meaning only one of the lists can be populated at a time. If a result produces multiple
+ * frames (used to transmit large results), all results will be of the same type (relational vs. graph) and contain the
+ * same element type (nodes, edges or paths).
+ */
 message GraphFrame {
+  // Field used to represent a set of nodes as part of a result. If this field is populated all others must not be set.
+  repeated ProtoNode nodes = 1;
+  // Field used to represent a set of edges as part of a result. If this field is populated all others must not be set.
+  repeated ProtoEdge edges = 2;
+  // Field used to represent a set of paths as part of a result. If this field is populated all others must not be set.
+  repeated ProtoPath paths = 3;
```

Comment:
Can you use a `oneof` here to ensure only one field is set?

Reply:
Because paths are returned as a mixed collection of edges and nodes, we need to be able to send both at the same time. What can be removed though is the paths field which is not needed.